### PR TITLE
ci: Use next tag for latest commit builds instead of nightly daily builds

### PR DESCRIPTION
**Pull request by mkuznyetsov**
Signed-off-by: Mykhailo Kuznietsov <mkuznets@redhat.com>

### What does this PR do?
Use next tag for doing nightly builds (create a duplicated job, so that we can retire the nightly tag separately when we are ready)
Replace occurances of "nightly" with "next"

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/19291

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: CONTRIBUTE.md [hunk 1]
```diff
@@ -343,7 +343,7 @@ The job names are `surge(alpine)` and `surge(rhel)`. Clicking on `Details` in th
 For example, alpine build of PR 805 would be hosted at: https://pr-check-805-alpine-che-plugin-registry.surge.sh/
 
 ### Nightly
```
Comment (nickboldt) [inline CONTRIBUTE.md:345]: this title probably needs an update :D

File: tools/build/src/che-plugin/che-plugins-meta-yaml-generator.ts [hunk 2]
```diff
@@ -36,7 +36,7 @@ export class ChePluginsMetaYamlGenerator {
         const metaId = `${publisher}/${name}`;
         const version = splitIds[2];
         let disableLatest: boolean;
-        // disable latest alias if version is not a number (probably like next/nightly/etc.)
+        // disable latest alias if version is not a number (probably like next/next/etc.)
```
Comment (nickboldt) [inline tools/build/src/che-plugin/che-plugins-meta-yaml-generator.ts:39]: this comment doesn't make sense after s/nightly/next; please remove the second next :D

File: tools/build/src/editor/che-editors-meta-yaml-generator.ts [hunk 3]
```diff
@@ -41,7 +41,7 @@ export class CheEditorsMetaYamlGenerator {
         const metaId = `${publisher}/${name}`;
         const version = splitIds[2];
         let disableLatest: boolean;
-        // disable latest alias if version is not a number (probably like next/nightly/etc.)
+        // disable latest alias if version is not a number (probably like next/next/etc.)
```
Comment (nickboldt) [inline tools/build/src/editor/che-editors-meta-yaml-generator.ts:44]: remove second next/ -- only need to provide the example once.

**Review by nickboldt — APPROVED** [review]
some minor tweaks needed to the docs/comments, but otherwise LGTM.

File: che-editors.yaml [hunk 4]
```diff
@@ -109,7 +109,7 @@ editors:
         volume: {}
       - name: che-machine-exec
         container:
-          image: 'quay.io/eclipse/che-machine-exec:nightly'
+          image: 'quay.io/eclipse/che-machine-exec:next'
```
Comment (ericwill) [inline che-editors.yaml:112]: Has this change already been made in the che-machine-exec side of things?

File: che-plugins.yaml [hunk 5]
```diff
@@ -36,7 +36,7 @@ plugins:
         image: 'quay.io/eclipse/che-buildah-base:1.14.0-b0c91d3'
         mountSources: true
         memoryLimit: 1Gi
-  - id: eclipse/che-async-pv-plugin/nightly
+  - id: eclipse/che-async-pv-plugin/next
```
Comment (ericwill) [inline che-plugins.yaml:39]: This is a plugin version, I don't think it should be changed here

File: tools/build/tests/_data/meta/machine-exec-plugin-meta.yaml [hunk 6]
```diff
@@ -32,7 +32,7 @@ spec:
         cookiesAuthEnabled: true
   containers:
     - name: che-machine-exec
-      image: 'quay.io/eclipse/che-machine-exec:nightly'
+      image: 'quay.io/eclipse/che-machine-exec:next'
```
Comment (ericwill) [inline tools/build/tests/_data/meta/machine-exec-plugin-meta.yaml:35]: Same question as earlier re: che-machine-exec version

File: che-editors.yaml (hunk 4 above)
Comment (mkuznyetsov) [reply che-editors.yaml:112]: yes, machine exec is now being published along with nightly

File: che-plugins.yaml (hunk 5 above)
Comment (mkuznyetsov) [reply che-plugins.yaml:39]: alright, keeping nightly

**Review by ericwill — APPROVED** [review]

**Pull request merged**